microsoft-integration/collaborative-documents/src/main/java/org/sakaiproject/microsoft/collaborativedocuments/controller/MainController.java

```diff
@@ -0,0 +1,620 @@
+	}
+
+	@GetMapping(value = {"/", "/index"})
+	public String showIndex(Model model, HttpServletRequest req) throws MicrosoftGenericException {
+		model.addAttribute("sortBy", StringUtils.isNotBlank(collaborativeDocumentsSessionBean.getLastSortBy()) ? collaborativeDocumentsSessionBean.getLastSortBy() :  SORT_BY_NAME + SORT_ASCENDING);
+		
+		String userId = sakaiProxy.getCurrentUserId();
+		
+		try {
+			//check delegated client -> will throw an exception if access token is invalid
+			microsoftAuthorizationService.checkDelegatedClient(userId);
+		}catch(MicrosoftInvalidTokenException e) {
+			//our configured access token is not valid. Try to get a new one through the automatic authorization process 
+			//(automatic: no confirmation screen will be shown)
+			return sendMicrosoftAuthorizationRedirect(true, req);
+		}catch(MicrosoftNoCredentialsException e) {
+			//this means there is no access token configured. Do nothing, continue and show the "configure" button.
+		}catch(MicrosoftCredentialsException e) {
+			//unexpected error. Remove current access token and let the process starts from the beginning
+			microsoftAuthorizationService.revokeAccessToken(userId);
+		}
+		
+		MicrosoftAccessToken mcAccessToken = microsoftAuthorizationService.getAccessToken(userId);
+		model.addAttribute("delegatedClientConfigured", (mcAccessToken != null));
+		if(mcAccessToken != null) {
+			model.addAttribute("mcUserAccount", mcAccessToken.getMicrosoftUserId());
+		}
+		
+		return INDEX_TEMPLATE;
+	}
+	
+	@GetMapping(value = {"/permissions"})
+	public String showPermissions(Model model) throws MicrosoftGenericException {
+		
+		return PERMISSIONS_TEMPLATE;
+	}
+	
+	/**
+	 * Called by AJAX - returns FRAGMENT
+	 * @param itemId
+	 * @param model
+	 * @return FRAGMENT
+	 * @throws MicrosoftGenericException
+	 */
+	@GetMapping(value = {"/items"})
+	public String loadItems(
+			@RequestParam(required = false) String teamId,
+			@RequestParam(required = false) String itemId,
+			@RequestParam(defaultValue = "name:0") String sortBy,
+			Model model
+	) throws MicrosoftGenericException {
+		
+		model.addAttribute("sortBy", sortBy);
+		collaborativeDocumentsSessionBean.setLastSortBy(sortBy);
+		
+		//first load
+		if(teamId == null) {
+			if(collaborativeDocumentsSessionBean.getItemsByTeam().isEmpty()) {
+				//get teams linked with this site
+				List<SiteSynchronization> ssList = microsoftSynchronizationService.getSiteSynchronizationsBySite(sakaiProxy.getCurrentSiteId());
+				for(SiteSynchronization ss : ssList) {
+					//check if team exists
+					MicrosoftTeam team = microsoftCommonService.getTeam(ss.getTeamId());
+					
+					if(team != null) {
+						collaborativeDocumentsSessionBean.getItemsByTeam().put(ss.getTeamId(), MicrosoftTeamWrapper.builder(team).build());
+					}
+				}
+				//first Team will be expanded by default, so we need to load it 
+				Optional<String> firstKey = collaborativeDocumentsSessionBean.getSortedTeamKeys().stream().findFirst();
```

Comment:
We can create a variable with
List<String> sortedTeamKeys = collaborativeDocumentsSessionBean.getSortedTeamKeys()

and then use it here
https://github.com/sakaiproject/sakai/pull/11760/files#diff-e565463da09aa357a8a7b6b1343c189b5ac26cf6250a679ea435e750f2e48df1R317

Reply:
Done